server/api.go

```diff
@@ -86,8 +86,8 @@ func NewAPI(p *Plugin, store store.Store) *API {
 
 // returnJSON writes the given data as json with the provided httpStatus
 func (a *API) returnJSON(w http.ResponseWriter, data any) {
-	w.WriteHeader(http.StatusOK)
```

Comment:
From https://pkg.go.dev/net/http:

>  Changing the header map after a call to [ResponseWriter.WriteHeader] has no effect unless the HTTP status code was of the 1xx class or the modified headers are trailers.

I think I broke this a while back when fixing a different endpoint.